Build failed:
third_party/tensorflow/compiler/tf2tensorrt/convert/weights.cc:61:11: error: enumeration value 'kBOOL' not handled in switch [-Werror,-Wswitch]
  switch (type_) {
          ^~~~~
1 error generated.

third_party/tensorflow/compiler/tf2tensorrt/convert/weights.cc:61:11: error: enumeration value 'kBOOL' not handled in switch [-Werror,-Wswitch]
  switch (type_) {
          ^~~~~
1 error generated.

--- a/tensorflow/compiler/tf2tensorrt/common/utils.cc
+++ b/tensorflow/compiler/tf2tensorrt/common/utils.cc
@@ -222,6 +222,11 @@ std::ostream& operator<<(std::ostream& os, const nvinfer1::DataType& v) {
     case nvinfer1::DataType::kBOOL:
       os << "kBOOL";
       break;
+#if IS_TRT_VERSION_GE(8, 5, 0, 0)
+    case nvinfer1::DataType::kUINT8:
+      os << "kUINT8";
+      break;
+#endif
   }
   return os;
 }

--- a/tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc
+++ b/tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc
@@ -123,6 +123,9 @@ const char* LayerTypeToString(nvinfer1::LayerType layer_type) {
     ADD_LAYER(REDUCE)
     ADD_LAYER(TOPK)
     ADD_LAYER(GATHER)
+#if IS_TRT_VERSION_GE(8, 5, 0, 0)
+    ADD_LAYER(GRID_SAMPLE)
+#endif
     ADD_LAYER(MATRIX_MULTIPLY)
     ADD_LAYER(RAGGED_SOFTMAX)
     ADD_LAYER(CONSTANT)
@@ -142,7 +145,24 @@ const char* LayerTypeToString(nvinfer1::LayerType layer_type) {
 #if IS_TRT_VERSION_GE(8, 0, 0, 0)
     ADD_LAYER(QUANTIZE)
     ADD_LAYER(DEQUANTIZE)
-#else
+#endif
+#if IS_TRT_VERSION_GE(8, 2, 0, 0)
+    ADD_LAYER(CONDITION)
+    ADD_LAYER(CONDITIONAL_INPUT)
+    ADD_LAYER(CONDITIONAL_OUTPUT)
+    ADD_LAYER(SCATTER)
+    ADD_LAYER(EINSUM)
+    ADD_LAYER(ASSERTION)
+#endif
+#if IS_TRT_VERSION_GE(8, 5, 0, 0)
+    ADD_LAYER(ONE_HOT)
+    ADD_LAYER(NON_ZERO)
+    ADD_LAYER(NMS)
+#endif
+#if IS_TRT_VERSION_GE(8, 6, 0, 0)
+    ADD_LAYER(REVERSE_SEQUENCE)
+#endif
+#if !IS_TRT_VERSION_GE(8, 0, 0, 0)
     // The TRT IRNNv2Layer has been deprecated in favor of the loop API.
     ADD_LAYER(RNN)
 #endif

--- a/tensorflow/compiler/tf2tensorrt/convert/utils.cc
+++ b/tensorflow/compiler/tf2tensorrt/convert/utils.cc
@@ -47,6 +47,8 @@ string DebugString(const DataType tf_type) {
       return "DT_INT8";
     case DT_BOOL:
       return "DT_BOOL";
+    case DT_UINT8:
+      return "DT_UINT8";
     default:
       return "Unknow TF DataType";
   }
@@ -64,6 +66,10 @@ string DebugString(const nvinfer1::DataType trt_dtype) {
       return "kINT32";
     case nvinfer1::DataType::kBOOL:
       return "kBOOL";
+#if IS_TRT_VERSION_GE(8, 5, 0, 0)
+    case nvinfer1::DataType::kUINT8:
+      return "kUINT8";
+#endif
     default:
       return "Invalid TRT data type";
   }
@@ -166,6 +172,11 @@ Status TfTypeToTrtType(DataType tf_type, nvinfer1::DataType* trt_type) {
       *trt_type = nvinfer1::DataType::kBOOL;
       break;
 #endif
+#if IS_TRT_VERSION_GE(8, 5, 0, 0)
+    case DT_UINT8:
+      *trt_type = nvinfer1::DataType::kUINT8;
+      break;
+#endif
     default:
       return errors::InvalidArgument("Unsupported tensorflow data type ",
                                      DataTypeString(tf_type));
@@ -187,6 +198,11 @@ Status TrtTypeToTfType(nvinfer1::DataType trt_type, DataType* tf_type) {
 #if IS_TRT_VERSION_GE(8, 2, 0, 0)
     case nvinfer1::DataType::kBOOL:
       *tf_type = DT_BOOL;
+      break;
+#endif
+#if IS_TRT_VERSION_GE(8, 5, 0, 0)
+    case nvinfer1::DataType::kUINT8:
+      *tf_type = DT_UINT8;
       break;
 #endif
     default:

--- a/tensorflow/compiler/tf2tensorrt/convert/weights.cc
+++ b/tensorflow/compiler/tf2tensorrt/convert/weights.cc
@@ -66,6 +66,9 @@ size_t TRT_ShapedWeights::size_bytes() const {
     case nvinfer1::DataType::kHALF:
       data_type_size = 2;
       break;
+#if IS_TRT_VERSION_GE(8, 5, 0, 0)
+    case nvinfer1::DataType::kUINT8:
+#endif
     case nvinfer1::DataType::kINT8:
     case nvinfer1::DataType::kBOOL:
       data_type_size = 1;

--- a/tensorflow/compiler/tf2tensorrt/kernels/trt_engine_op.cc
+++ b/tensorflow/compiler/tf2tensorrt/kernels/trt_engine_op.cc
@@ -290,6 +290,9 @@ void* GetTensorAddress(const Tensor* tensor_ptr) {
 #if IS_TRT_VERSION_GE(8, 2, 0, 0)
     TYPECASE(DT_BOOL, tensor_ptr);
 #endif
+#if IS_TRT_VERSION_GE(8, 5, 0, 0)
+    TYPECASE(DT_UINT8, tensor_ptr);
+#endif
     default: {
       LOG(ERROR) << "Unsupported Data type " << DataTypeString(tensor_type);
       return nullptr;
@@ -1296,7 +1299,8 @@ Status TRTEngineOp::AllocateCalibrationResources(
     void* device_address = GetTensorAddress(input);
     if (device_address == nullptr) {
       return errors::InvalidArgument(
-          "Unsupported data type encountered in input ", i);
+          "Unsupported data type [", DebugString(t.dtype()),
+          "] encountered in input ", i);
     }
     cres->device_buffers_.emplace(
         StrCat(IONamePrefixes::kInputPHName, i),

--- a/tensorflow/compiler/tf2tensorrt/utils/trt_engine_utils.cc
+++ b/tensorflow/compiler/tf2tensorrt/utils/trt_engine_utils.cc
@@ -104,6 +104,11 @@ Status SetupBindings(nvinfer1::ICudaEngine* cuda_engine, const Tensor& tensor,
       buffers[binding_index] = const_cast<bool*>(tensor.flat<bool>().data());
       break;
 #endif
+#if IS_TRT_VERSION_GE(8, 5, 0, 0)
+    case nvinfer1::DataType::kUINT8:
+      buffers[binding_index] = const_cast<uint8*>(tensor.flat<uint8>().data());
+      break;
+#endif
     default:
       return errors::Internal("Unknown TRT data type: ",
                               static_cast<int>(dtype));

